x/gateway/module/tx_stake_gateway_test.go

```diff
@@ -0,0 +1,150 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/status"
+
+	"github.com/pokt-network/poktroll/testutil/network"
+	"github.com/pokt-network/poktroll/testutil/yaml"
+	gateway "github.com/pokt-network/poktroll/x/gateway/module"
+	"github.com/pokt-network/poktroll/x/gateway/types"
+)
+
+func TestCLI_StakeGateway(t *testing.T) {
+	net, _ := networkWithGatewayObjects(t, 2)
+	val := net.Validators[0]
+	ctx := val.ClientCtx
+
+	// Create a keyring and add an account for the gateway to be staked
+	kr := ctx.Keyring
+	accounts := testutil.CreateKeyringAccounts(t, kr, 1)
+	gatewayAccount := accounts[0]
+
+	// Update the context with the new keyring
+	ctx = ctx.WithKeyring(kr)
+
+	// Common args used for all requests
+	commonArgs := []string{
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastSync),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(net.Config.BondDenom, sdkmath.NewInt(10))).String()),
+	}
+
+	tests := []struct {
+		desc          string
+		address       string
+		inputConfig   string
+		expectedError *sdkerrors.Error
+	}{
+		{
+			desc:    "stake gateway: invalid address",
+			address: "invalid",
+			inputConfig: `
+			  stake_amount: 1000upokt
+				`,
+			expectedError: types.ErrGatewayInvalidAddress,
+		},
+		{
+			desc: "stake gateway: missing address",
+			// address:     gatewayAccount.Address.String(),
+			inputConfig: `
+			  stake_amount: 1000upokt
+				`,
+			expectedError: types.ErrGatewayInvalidAddress,
+		},
+		{
+			desc:    "stake gateway: invalid stake amount (zero)",
+			address: gatewayAccount.Address.String(),
+			inputConfig: `
+			  stake_amount: 0upokt
+				`,
+			expectedError: types.ErrGatewayInvalidStake,
+		},
+		{
+			desc:    "stake gateway: invalid stake amount (negative)",
+			address: gatewayAccount.Address.String(),
+			inputConfig: `
+			  stake_amount: -1000upokt
+				`,
+			expectedError: types.ErrGatewayInvalidStake,
+		},
+		{
+			desc:    "stake gateway: invalid stake denom",
+			address: gatewayAccount.Address.String(),
+			inputConfig: `
+			  stake_amount: 1000invalid
+				`,
+			expectedError: types.ErrGatewayInvalidStake,
+		},
+		{
+			desc:    "stake gateway: invalid stake missing denom",
+			address: gatewayAccount.Address.String(),
+			inputConfig: `
+			  stake_amount: 1000
+				`,
+			expectedError: types.ErrGatewayInvalidStake,
+		},
+		{
+			desc:          "stake gateway: invalid stake missing stake",
+			address:       gatewayAccount.Address.String(),
+			inputConfig:   ``,
+			expectedError: types.ErrGatewayInvalidStake,
+		},
+		{
+			desc:    "stake gateway: valid",
+			address: gatewayAccount.Address.String(),
+			inputConfig: `
+			  stake_amount: 1000upokt
+				`,
+		},
+	}
+
+	// Initialize the Gateway Account by sending it some funds from the validator account that is part of genesis
+	network.InitAccount(t, net, gatewayAccount.Address)
```

Comment:
Wdyt about moving this up closer to the `testutil.CreateKeyringAccounts()` call?